src/main/java/mate/academy/mybookshop/service/OrderServiceImpl.java

```diff
@@ -0,0 +1,106 @@
+package mate.academy.mybookshop.service;
+
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import lombok.RequiredArgsConstructor;
+import mate.academy.mybookshop.dto.order.CreateOrderRequestDto;
+import mate.academy.mybookshop.dto.order.OrderResponseDto;
+import mate.academy.mybookshop.dto.order.UpdateOrderRequestDto;
+import mate.academy.mybookshop.dto.orderitem.OrderItemResponseDto;
+import mate.academy.mybookshop.entity.CartItemEntity;
+import mate.academy.mybookshop.entity.OrderEntity;
+import mate.academy.mybookshop.entity.OrderItemEntity;
+import mate.academy.mybookshop.entity.UserEntity;
+import mate.academy.mybookshop.exception.EmptyCartException;
+import mate.academy.mybookshop.exception.EntityNotFoundException;
+import mate.academy.mybookshop.mapper.OrderItemMapper;
+import mate.academy.mybookshop.mapper.OrderMapper;
+import mate.academy.mybookshop.repository.CartItemRepository;
+import mate.academy.mybookshop.repository.OrderItemRepository;
+import mate.academy.mybookshop.repository.OrderRepository;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final OrderRepository orderRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final CartItemRepository cartItemRepository;
+    private final OrderItemMapper orderItemMapper;
+    private final OrderMapper orderMapper;
+
+    @Override
+    public OrderResponseDto createOrder(CreateOrderRequestDto requestDto, UserEntity user) {
+        Set<CartItemEntity> cartItems = cartItemRepository
+                .findCartItemEntitiesByShoppingCartId(user.getId());
+        if (cartItems.isEmpty()) {
+            throw new EmptyCartException("Cannot create order without items.");
+        }
+
+        OrderEntity order = orderMapper.toEntity(requestDto);
+        order.setOrderDate(LocalDateTime.now());
+        order.setUser(user);
+
+        order.setTotal(calculateTotal(cartItems));
+        order.setShippingAddress(orderMapper.toEntity(requestDto).getShippingAddress());
```

Comment:
duplicated call of `orderMapper.toEntity(requestDto)` and you do not need to set address manually, it is already mapped from the dto

Reply:
fixed